src/chipeitlib/src/main/java/es/chipeit/lib/core/InstructionSet.kt

```diff
@@ -46,42 +48,167 @@ internal fun ldVxByte(instruction: Int, registers: IRegisters) {
 // 8xy3 - XOW Vx, Vy
 
 // 8xy4 - ADD Vx, Vy
+internal fun addVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vx + vy
+
+    registers.v[0xF] = (add shr 8).toByte()
+    registers.v[x] = add.toByte()
+
+    registers.pc += 2
+}
 
 // 8xy5 - SUB Vx, Vy
+internal fun subVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vx - vy
+
+    registers.v[x] = add.toByte()
+    registers.v[0xF] = if (add > 0) 1 else 0
+
+    registers.pc += 2
+}
 
 // 8xy6 - SHR Vx {, Vy}
 
 // 8xy7 - SUBN Vx, Vy
+internal fun subnVxVy(instruction: Int, registers: IRegisters) {
+    val x = instruction shr 2 * 4 and 0xF
+    val y = instruction shr 1 * 4 and 0xF
+
+    var vx = registers.v[x].toInt() and 0xFF
+    var vy = registers.v[y].toInt() and 0xFF
+
+    val add = vy - vx
+
+    registers.v[0xF] = if (add > 0) 1 else 0
```

Comment:
The order of `v[x]` and `v[0xF]` doesn't match the other instructions that basically do the same.